mlir-sycl/doc/design/SYCLGridRepresentation.md

```diff
@@ -98,110 +91,25 @@ file also generates call to register it with no further action.
-retrieved using `SYCLDialect::lookupMethodDefinition(llvm::StringRef,
-mlir::FunctionType)`.
-
-In the codegen side, when code is generated for a function in the `sycl`
-namespace (an `mlir::func::FuncOp` is obtained), if this corresponds to an
-operation, it is inserted in the SYCL dialect. After this insertions takes
-place, only declarations can be overriden by functions of the same name.
-
 ### Lowering
 
-All operations implementing `SYCLMethodOpInterface` are lowered to a `sycl.call`
-operation. In order to do so, the corresponding attributes of the operation are
-used to construct the `sycl.call`. Regarding the `MangledFunctionName`
-attribute, this was not needed in the original operation as the lowering process
-makes use of the definition register to obtain the function implementing the
-operation, insert it in the module if it is not present yet, and use the name of
-this function to build the attribute.
-
-In order to create a legal function call, the `FunctionArgTypes` attribute is
-used to perform the necessary transformations beforehand:
-
-1. Build a `memref.alloca()` of type `memref<1xTy>`, being `Ty` the type of the
-   first argument;
-2. Store the value there;
-3. If the argument type is a `memref` and the shapes differ, introduce a
-   `memref.cast` operation;
-4. If the memory spaces differ, use a `memref.memory_space_cast` to cast
-   to the required one;
-5. If needed, cast to the base type using `sycl.cast` (or other operation we
-   might have in the future).
-
-In case the operation to be lowered does not have a definition in the register,
-it will be "decanonicalized", following the inverse process to the 
-canonicalization process explained above and each operation being generated will
-be lowered following the same process.
-
-If an operation does not count with a definition in the register and cannot be
-split into further operations, the lowering process will fail, as the user would
-have failed to provide a definition for an operation.
-
-#### Lowering example
-
-The following `sycl.nd_item.get_global_id` operation:
-
-```MLIR
-%1 = sycl.nd_item.get_global_id(%nd, %idx) {FunctionArgTypes = {!llvm.ptr<struct<...>, 4>}, FunctionName = @get_global_id, TypeName = @nd_item} : (!sycl_nd_item_1_, i32) -> i64
-```
-
-Would be lowered to:
-
-```MLIR
-%1 = memref.alloca() : memref<1x!sycl_nd_item_1_>
-memref.store %nd, %1[0] : memref<1x!sycl_nd_item_1_>
-%2 = memref.memory_space_cast %1 : memref<1x!sycl_nd_item_1_> to memref<1x!sycl_nd_item_1_, 4>
-%3 = sycl.cast(%2) : (memref<1x!sycl_nd_item_1_, 4>) -> !llvm.ptr<struct<...>, 4>
-%4 = sycl.call(%3, %idx) {FunctionName = @get_global_id, MangledFunctionName = <MFN>, TypeName = @nd_item} : (!llvm.ptr<struct<...>, 4>, i32) -> i64
-```
-
-Being `<MFN>` the name of the function to call, retrieved by calling
-`SYCLDialect::lookupMethodDefinition`.
+Lowering of these operations mimics the homologous function implementations in
+[the SYCL headers](../../../sycl/include).
```

Comment:
Can you elaborate slightly here, e.g. which dialects we lower into?

Reply:
Extended a bit